ide/editor.lib/src/org/netbeans/lib/editor/hyperlink/spi/HyperlinkLocation.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.lib.editor.hyperlink.spi;
+
+import java.util.Arrays;
+import java.util.Objects;
+import org.openide.filesystems.FileObject;
+
+/**
+ * Represents hyperlink target location.
+ *
+ * @author Dusan Balek
+ * @since 4.20
+ */
+public final class HyperlinkLocation {
+
+    private final FileObject fileObject;
+    private final int[] range;
+
+    public HyperlinkLocation(FileObject fileObject, int[] range) {
+        this.fileObject = fileObject;
+        this.range = range;
+    }
+
+    /**
+     * Get target file object.
+     *
+     * @return file object
+     */
+    public FileObject getFileObject() {
+        return fileObject;
+    }
+
+    /**
+     * Returns target's range within its file object.
+     *
+     * @return span of the target
+     */
+    public int[] getRange() {
```

Comment:
Can you document what's the meaning of range?

Reply:
`getStartOffset()` and `getEndOffset()` methods used instead of `getRange()`